File: CSharp/Controllers/JsonController.cs
using System.IO;
using Newtonsoft.Json;
using System;
using Newtonsoft.Json.Linq;
using System.Collections;

namespace SharpFileWorker
{
    class JsonController
    {
        public static void DeleteFile(String path)
        {
            FileController.DeleteFile(path);
        }
        public static void ReadFile(String path)
        {
            if (!File.Exists(path))
            {
                throw new FileNotFoundException($"File with path: {path} not found!");
            }
            string text = File.ReadAllText(path);
            JObject jsonObject = JsonConvert.DeserializeObject<JObject>(text);
            Console.WriteLine(JsonConvert.SerializeObject(jsonObject, Formatting.Indented));
        }
        public static void CreateTemplateJson(String path)
        {
            if (File.Exists(path))
            {
                throw new FileNotFoundException($"File with path: {path} already exists!");
            }
            JObject jsonObject = new JObject();
            jsonObject["id"] = 1;
            jsonObject["name"] = "some default name";
            jsonObject["roles"] = new JArray(new String[] { "ADMIN", "MODERATOR" });
            File.WriteAllText(path, JsonConvert.SerializeObject(jsonObject, Formatting.Indented));
        }
        public static void CreateJsonObject(String path, params String[] entries)
        {
            if (File.Exists(path))
            {
                throw new FileNotFoundException($"File with path: {path} already exists!");
            }
            JObject jsonObject = (JObject) JsonConvert.DeserializeObject(String.Join('\n', entries));
            File.WriteAllText(path, JsonConvert.SerializeObject(jsonObject, Formatting.Indented));
        }
        public static Argument[] GetArguments()
        {
            Argument deleteJson = Argument
                .CreateArgument()
                .SetNames("-dj", "--delete-json")
                .SetDescription("Delete json file")
                .SetArgumentName("JSON_FILE_NAME")
                .SetArgsCount(1)
                .SetAction((args) => JsonController.DeleteFile(args[0]));
            Argument readJson = Argument
                .CreateArgument()
                .SetNames("-rj", "--read-json")
                .SetDescription("Read json file by provided path")
                .SetArgumentName("JSON_FILE_NAME")
                .SetArgsCount(1)
                .SetAction((args) => JsonController.ReadFile(args[0]));
            Argument createTemplate = Argument
                .CreateArgument()
                .SetNames("-ctj", "--create-template-json")
                .SetDescription("Create template json by provided path")
                .SetArgumentName("JSON_FILE_NAME")
                .SetArgsCount(1)
                .SetAction((args) => JsonController.CreateTemplateJson(args[0]));
            Argument create = Argument
                .CreateArgument()
                .SetNames("-cj", "--create-json")
                .SetDescription("Create json by provided path and provided entries")
                .SetArgumentName("JSON_FILE_NAME")
                .SetArgsCount(2)
                .SetAction((args) => JsonController.CreateJsonObject(args[0], args.SubArray(1, args.Length - 1)));
            return new Argument[]
            {
                createTemplate,
                create,
                readJson,
                deleteJson
            };
        }
    }
}

File: python/json_controller.py
from files import file_exists
from file_controller import delete_file
import json

def delete_json (path: str) -> None:
    delete_file(path)

def read_file(path: str) -> None:
    if not file_exists(path):
        raise Exception(f'File with name {path} not found')
    with open(path) as file:
        print(json.load(file))

def create_template_json(path: str) -> None:
    if file_exists(path):
        raise Exception(f'File with name {path} already exists')
    template = {
        'id': 1,
        'name': 'Ivan',
        'roles': ['ADMIN', 'MODERATOR']
    }
    with open(path, 'w') as file:
        json.dump(template, file)

def create_json_object(path: str, *data: 'list[str]') -> None:
    if file_exists(path):
        raise Exception(f'File with name {path} already exists')
    try:
        curr_json = json.loads(''.join(data))
        with open(path, 'w') as file:
            json.dump(curr_json, file)
    except:
        print('Error while reading json')

controller_schema = {
    'read_json' : read_file,
    'delete_json': delete_json,
    'create_template_json': create_template_json,
    'create_json': create_json_object,
}